java/cdk/src/main/java/sleeper/cdk/stack/TransactionLogSnapshotCreationStack.java

```diff
@@ -0,0 +1,136 @@
+                .timeout(Duration.minutes(1))
+                .logGroup(createLambdaLogGroup(this, "TransactionLogSnapshotCreationLogGroup", creationFunctionName, instanceProperties)));
+
+        Rule rule = Rule.Builder.create(this, "TransactionLogSnapshotCreationSchedule")
+                .ruleName(SleeperScheduleRule.TRANSACTION_LOG_SNAPSHOT_CREATION.buildRuleName(instanceProperties))
+                .schedule(Schedule.rate(Duration.minutes(
+                        instanceProperties.getLong(TRANSACTION_LOG_SNAPSHOT_CREATION_LAMBDA_PERIOD_IN_MINUTES))))
+                .targets(List.of(new LambdaFunction(snapshotCreationTrigger)))
+                .enabled(!shouldDeployPaused(this))
+                .build();
+        instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_CREATION_RULE, rule.getRuleName());
+
+        Queue deadLetterQueue = Queue.Builder
+                .create(this, "TransactionLogSnapshotDeadLetterQueue")
+                .queueName(String.join("-", "sleeper", instanceProperties.get(ID), "TransactionLogSnapshotDLQ.fifo"))
+                .fifo(true)
+                .build();
+        Queue queue = Queue.Builder
+                .create(this, "TransactionLogSnapshotQueue")
+                .queueName(String.join("-", "sleeper", instanceProperties.get(ID), "TransactionLogSnapshotQ.fifo"))
+                .deadLetterQueue(DeadLetterQueue.builder()
+                        .maxReceiveCount(1)
+                        .queue(deadLetterQueue)
+                        .build())
+                .fifo(true)
+                .visibilityTimeout(Duration.seconds(70))
+                .build();
+        instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_CREATION_QUEUE_URL, queue.getQueueUrl());
+        instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_CREATION_QUEUE_ARN, queue.getQueueArn());
+        instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_CREATION_DLQ_URL, deadLetterQueue.getQueueUrl());
+        instanceProperties.set(TRANSACTION_LOG_SNAPSHOT_CREATION_DLQ_ARN, deadLetterQueue.getQueueArn());
+        createAlarmForDlq(this, "TransactionLogSnapshotCreationAlarm",
+                "Alarms if there are any messages on the dead letter queue for the transaction log snapshot creation queue",
+                deadLetterQueue, topic);
+        errorMetrics.add(Utils.createErrorMetric("Transaction Log Snapshot Errors", deadLetterQueue, instanceProperties));
+        queue.grantSendMessages(snapshotCreationTrigger);
+
+        snapshotCreationLambda.addEventSource(SqsEventSource.Builder.create(queue)
+                .batchSize(instanceProperties.getInt(TRANSACTION_LOG_SNAPSHOT_CREATION_BATCH_SIZE)).build());
+
+        coreStacks.grantReadTablesStatus(snapshotCreationTrigger);
+        coreStacks.grantInvokeScheduled(snapshotCreationTrigger, queue);
+        coreStacks.grantReadTablesMetadata(snapshotCreationLambda);
+        transactionLogStateStoreStack.grantReadWriteSnapshots(snapshotCreationLambda);
```

Comment:
It looks like grantReadTablesStatus should be enough rather than grantReadTablesMetadata, when we're already telling the state store stack directly to set up the grants that are needed there?